scipy/spatial/_hausdorff.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (C)  Tyler Reddy, Richard Gowers, and Max Linke, 2016
+#
+# Distributed under the same BSD license as Scipy.
+#
+# adapted from scipy's cython version
```

Comment:
Isn't the discussion above now pointing to a less drastic approach where `# cython: np_pythran=True` is used in our current Cython version of the algorithm, instead of having two source files with a similar but slightly modified algorithm? The two source files for one algorithm approach seems to have a non-trivial maintenance burden in my view. See also https://github.com/scipy/scipy/pull/10020 which tries to modify low-level stuff in the algorithm, but hard to find reviewers for.

If the Cython implementation also has to be somewhat re-written to make it suitable for Pythran support / speed-up as implied, I'd suggest that it should remain at least as fast as the unmodified Cython version prior to modification when used without Pythran.

This feels constraining though--is that really reasonable to ask?

Reply:
I can understand this option, especially as it would avoid duplicate code base. the pythran mode of cython is not as well tested as pythran as a whole, but that's definitively an option.